sdk/python/tests/test_historical_retrieval.py

```diff
@@ -547,7 +588,26 @@ def test_historical_features_from_bigquery_sources(
             .reset_index(drop=True),
             check_dtype=False,
         )
+
         table_from_df_entities = job_from_df.to_arrow()
         assert_frame_equal(
             actual_df_from_df_entities, table_from_df_entities.to_pandas()
         )
+
+
+@pytest.mark.integration
+def test_feature_name_collision_on_historical_retrieval():
+
+    # _validate_feature_refs is the function that checks for colliding feature names
+    with pytest.raises(FeatureNameCollisionError):
```

Comment:
It'd be good we can assert the contents of the error message and that it points to the right column that has collisions.

Reply:
@achals Done! Thanks for the review, I added assertion for each of the two possible error messages alongside the other changes you requested. Hopefully it's good to go now.